python/setup.py

```diff
@@ -35,59 +42,143 @@
     cuda_include_dir = os.path.join(os.environ.get('CUDA_HOME'), 'include')
 
 
-rmm_include_dir = '/include'
-rmm_lib_dir = '/lib'
-
-# including cumlprims dirs, in case rmm paths change in the future
-cumlprims_include_dir = '/include'
-cumlprims_lib_dir = '/lib'
+conda_lib_dir = os.path.normpath(sys.prefix) + '/lib'
+conda_include_dir = os.path.normpath(sys.prefix) + '/include'
 
 if os.environ.get('CONDA_PREFIX', None):
     conda_prefix = os.environ.get('CONDA_PREFIX')
-    rmm_include_dir = conda_prefix + rmm_include_dir
-    rmm_lib_dir = conda_prefix + rmm_lib_dir
+    conda_include_dir = conda_prefix + '/include'
+    conda_lib_dir = conda_prefix + '/lib'
+
+
+##############################################################################
+# - Subrepo checking and cloning ---------------------------------------------
+
+
+def clone_repo(name, GIT_REPOSITORY, GIT_TAG, first=False):
+    """
+    Function to clone repos in case they were not cloned by cmake.
+    Variables are named identical to the cmake counterparts for clarity,
+    in spite of not being very pythonic.
+    """
+    if first:
+        subprocess.check_call(['rm', '-rf', 'external'])
+
+    subprocess.check_call(['git', 'clone',
+                           GIT_REPOSITORY,
+                           'external/' + name])
+    wd = os.getcwd()
+    os.chdir("external/" + name)
+    subprocess.check_call(['git', 'checkout',
+                          GIT_TAG])
+    os.chdir(wd)
+
+
+if "clean" not in sys.argv:
+    if os.environ.get('CUML_BUILD_PATH', False):
+        libcuml_path = '../' + os.environ.get('CUML_BUILD_PATH')
```

Comment:
If I'm just building in the same place repeatedly, do I need to set this variable so I don't have to clone each time?
What ensures that CUML_BUILD_PATH has the correct git tags? Can we check that somehow?

Reply:
It ended up being a nightmare to check the tags. Checking the commit is fine, but given that we might prefer to use git tags (like I did for CUB and CUTLASS already in CMakeLists) instead of restricting cmake to only being able to clone commit hashes, this ended up being a git-hell, so for now the best we can do is to check the existence of the folder :/